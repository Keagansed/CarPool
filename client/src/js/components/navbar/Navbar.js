import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Navbar extends Component{

    render(){
        return(
            <div className="navbar-component navbarBottom">
                <div className="container-fluid p-0">
                    <div className="row">
                        <div className="col-md-12 ">
                            <ul className="nav nav-pills navbarBottom text-center">
                                <li className="nav-item ml-auto w-25">
                                    <Link to={`/`} className="nav-link no-hover">
                                        <i className="fa fa-cog fa-2x text-secondary d-inline"></i>&nbsp;
                                    </Link>
                                </li>
                                <li className="nav-item w-25">
                                    <Link to={`/HomePage`} className="nav-link no-hover" >
                                        <i className="fa fa-fw fa-car fa-2x d-inline text-secondary"></i>
                                    </Link>
                                </li>
                                <li className="nav-item mr-auto text-secondary w-25">
                                <Link to={{
                                        pathname: "/HomePage" + this.props.token,
                                        state: { token: this.props.token }
                                    }} className="nav-link no-hover"
                                >
                                    <i className="fa fa-fw fa-2x fa-user d-inline text-secondary"></i>
                                </Link>
                                </li>
                            </ul>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Navbar