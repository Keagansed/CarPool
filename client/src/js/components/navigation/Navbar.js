// File Type: Component

import React, { Component } from 'react';
import { Link } from 'react-router-dom';

/*
* The purpose of the Navbar class is to provide a navbar that is available throughout the application.
* This Navbar provides links to the home page, the user's profile page, and the settings page.
*/
class Navbar extends Component {

    /*
    * The purpose of the render method is to enable the rendering of this component.
    * It returns react elements using JSX.
    */
    render() {
        return(
            <div className="height-50px fixed-bottom container-fluid pad-10px bg-aqua">
                <div className="row">
                    <div className="col-4 txt-center">
                        <Link to={`/Settings`}>
                            <i className="fa fa-cog txt-purple txt-30px"></i>
                        </Link>
                    </div>
                    <div className="col-4 txt-center">
                        <Link to={`/HomePage`}>
                            <i className="fa fa-home txt-purple txt-30px"></i>
                        </Link>
                    </div>
                    <div className="col-4 txt-center">
                        <Link to={{pathname: "/ProfilePage/" + this.props.token, state: { token: this.props.token }}}>
                            <i className="fa fa-user txt-purple txt-30px"></i>
                        </Link>
                    </div>
                </div>
            </div>
        )
    }
}

export default Navbar;