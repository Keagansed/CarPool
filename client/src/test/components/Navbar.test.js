import React from 'react';
import Navbar from '../../js/components/navigation/Navbar';
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('Navbar Component', () => {
    let container;

    beforeEach(() => {
        container = shallow(<MemoryRouter><Navbar /></MemoryRouter>);
    });

    it('captures snapshot', () => {
        const wrapper = renderer.create(<MemoryRouter><Navbar /></MemoryRouter>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly', () => {
        expect(container.length).toEqual(1);
    });
});