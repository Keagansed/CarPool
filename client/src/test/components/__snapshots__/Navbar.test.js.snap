// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component captures snapshot 1`] = `
<div
  className="height-50px fixed-bottom container-fluid pad-10px bg-aqua"
>
  <div
    className="row"
  >
    <div
      className="col-4 txt-center"
    >
      <a
        href="/Settings"
        onClick={[Function]}
      >
        <i
          className="fa fa-cog txt-purple txt-30px"
        />
      </a>
    </div>
    <div
      className="col-4 txt-center"
    >
      <a
        href="/HomePage"
        onClick={[Function]}
      >
        <i
          className="fa fa-home txt-purple txt-30px"
        />
      </a>
    </div>
    <div
      className="col-4 txt-center"
    >
      <a
        href="/ProfilePage/undefined"
        onClick={[Function]}
      >
        <i
          className="fa fa-user txt-purple txt-30px"
        />
      </a>
    </div>
  </div>
</div>
`;
